QuizApp/QuizApp/UI/Quiz/QuizViewModel.swift

```diff
@@ -23,11 +23,23 @@ class QuizViewModel {
     func loadQuizes(for category: QuizCategory) {
         Task(priority: .background) {
             do {
-                self.quizes = try await quizUseCase
-                    .getQuizes(for: QuizCategoryModel(rawValue: category.rawValue)!)
+                var quizes: [QuizModel] = []
+
+                if category != .all {
+                    quizes = try await quizUseCase.getQuizes(for: QuizCategoryModel(rawValue: category.rawValue)!)
+                } else {
+                    quizes = try await quizUseCase.getQuizes()
+                }
+
+                self.quizes = quizes
                     .map {
                         Quiz(from: $0)
                     }
+                    .reduce([QuizCategory: [Quiz]]()) { (dict, quiz) -> [QuizCategory: [Quiz]] in
```

Comment:
Can you use https://developer.apple.com/documentation/swift/dictionary/init(grouping:by:) ?